### ISSUE TITLE
no longer compiles: inference error on crate `time` caused by an API change in Rust 1.80.0

### ISSUE THREAD

**Issue by zesty** [Sh1Yo/x8#59]
```
   Compiling data-encoding v2.3.3
   Compiling psl-types v2.0.11
   Compiling publicsuffix v2.2.3
   Compiling lru-cache v0.1.2
   Compiling thiserror-impl v1.0.40
error[E0282]: type annotations needed for `Box<_>`
  --> /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/time-0.3.20/src/format_description/parse/mod.rs:83:9
   |
83 |     let items = format_items
   |         ^^^^^
...
86 |     Ok(items.into())
   |              ---- type must be known at this point
   |
   = note: this is an inference error on crate `time` caused by an API change in Rust 1.80.0; update `time` to version `>=0.3.35` by calling `cargo update`

   Compiling tokio-macros v1.8.2
   Compiling futures-macro v0.3.27
   Compiling openssl-macros v0.1.0
For more information about this error, try `rustc --explain E0282`.
error: could not compile `time` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

--------------------

output of `rustc --explain E0282`:

Edit: SNIP (see next)

**Comment by zesty** [comment]
`cargo update` does fix this

**Comment by Sh1Yo** [comment]
Thanks for notifying me and providing a solution. Fixed.

**Comment by tobyweb3x** [comment]
i've tried `cargo update` and `cargo check` still says the same.
@zesty @Sh1Yo

**Comment by zesty** [comment]
perhaps not fixed in 1.80.0?

I'm on 1.85.0 (4d91de4e4 2025-02-17)

**Comment by Druthyn** [comment]
Same issue here. Output from cargo update and rust version below:

> $ cargo update
> warning: `/home/<user>/wezterm-20240203-110809-5046fc22/.cargo/config` is deprecated in favor of `config.toml`
> note: if you need to support cargo 1.38 or earlier, you can symlink `config` to `config.toml`
>     Updating crates.io index
>     Updating git repository `https://github.com/wez/xcb-imdkit-rs.git`
> error: failed to select a version for the requirement `wgpu = "^0.18"`
>   version 0.18.0 is yanked
> location searched: crates.io index
> required by package `wezterm-gui v0.1.0 (/home/<user>/wezterm-20240203-110809-5046fc22/wezterm-gui)`
> 

> $ rustc -V
> rustc 1.86.0 (05f9846f8 2025-03-31)
